x-pack/plugins/security_solution/server/lib/detection_engine/signals/send_telemetry_events.ts

```diff
@@ -11,15 +11,29 @@ import { BuildRuleMessage } from './rule_messages';
 import { SignalSearchResponse, SignalSource } from './types';
 import { Logger } from '../../../../../../../src/core/server';
 
-export interface SearchResultWithSource {
+interface SearchResultWithEventId {
+  _source: {
+    event: {
+      id: string;
+    };
+  };
+}
+
+interface SearchResultSource {
   _source: SignalSource;
 }
 
-export function selectEvents(filteredEvents: SignalSearchResponse): TelemetryEvent[] {
+type SearchResultWithSource = SearchResultSource & SearchResultWithEventId;
+
+export function selectEvents(
+  filteredEvents: SignalSearchResponse,
+  signalIdMap: Map<string, string>
+): TelemetryEvent[] {
   // @ts-expect-error @elastic/elasticsearch _source is optional
   const sources: TelemetryEvent[] = filteredEvents.hits.hits.map(function (
     obj: SearchResultWithSource
   ): TelemetryEvent {
+    obj._source.signal_id = signalIdMap.get(obj._source.event.id);
```

Comment:
You could be doing a lot of unnecessary work here, particularly if the security user isn't using the endpoint package, and using other detection rules. I'm also not sure how I feel about this mutation, particularly because `.event.id` isn't necessarily read-safe.

It might be intuitive to write this as:

```
  -> Collect Telemetry Events
  -> Filter for Endpoint Alerts
  -> Enrich with ancestor signal id, default null
  -> return
```

Reply:
Yeah that flow sounds better, I can do that.

Reply:
I think "ancestor signal id" is actually confusing/ maybe not semantically correct either since in signal parlance, a detection engine alert can have multiple ancestors (It's an array in the return). Do you think the name `signal_id` is sufficient or would you prefer another name?